src/app/components/LoadMoreProducts.tsx

```
@@ -10,142 +10,141 @@ import Link from 'next/link';
-              ...p,
-              images: p?.images?.map((img) => ({
-                ...img,
-                color: img.color ?? null,
-              })),
+  const [products, setProducts] = useState<
+    Product[] | FullProduct[]
+  >(
+    category === 'all'
+      ? [
+          ...hardCodedProducts,
+          ...initialProducts?.map((p) => ({
```

Biome (1.9.4)

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)